Overview
-------------------------

Here we list links and references to the external software used in our analyses.

cutadapt
-------------------------

> "Cutadapt removes adapter sequences from high-throughput sequencing reads".</br>
> Martin, EMBnet.journal, vol. 17, no. 1, p. 10, May 2011.</br>
> https://github.com/marcelm/cutadapt

EPA-ng
-------------------------

> "EPA-ng: Massively Parallel Evolutionary Placement of Genetic Sequences".</br>
> Barbera et al., Systematic Biology, 2018.</br>
> https://github.com/Pbdas/epa-ng

IQ-tree
-------------------------

> "IQ-TREE: A fast and effective stochastic algorithm for estimating maximum-likelihood phylogenies".</br>
> Nguyen et al., Mol. Biol. Evol., vol. 32, no. 1, pp. 268–74, Jan. 2015.</br>
> http://www.iqtree.org/

PaPaRa
-------------------------

> "PaPaRa 2.0: A Vectorized Algorithm for Probabilistic Phylogeny-Aware Alignment Extension".</br>
> Berger et al., Heidelberg, 2012.</br>
> https://sco.h-its.org/exelixis/web/software/papara/index.html

PEAR
-------------------------

> "PEAR: A fast and accurate Illumina Paired-End reAd mergeR".</br>
> Zhang et al., Bioinformatics, vol. 30, no. 5, pp. 614–620, 2014.</br>
> https://sco.h-its.org/exelixis/web/software/pear/

RAxML
-------------------------

> "RAxML version 8: a tool for phylogenetic analysis and post-analysis of large phylogenies".</br>
> Stamatakis, Bioinformatics, vol. 30, no. 9, pp. 1312–1313, 2014.</br>
> https://sco.h-its.org/exelixis/web/software/raxml/index.html

RAxML-ng
-------------------------

> "RAxML-NG: A fast, scalable, and user-friendly tool for maximum likelihood phylogenetic inference".</br>
> Kozlov et al., bioRxiv, 2018.</br>
> https://github.com/amkozlov/raxml-ng

Sativa
-------------------------

> "Phylogeny-aware identification and correction of taxonomically mislabeled sequences".</br>
> Kozlov et al., Nucleic Acids Res., vol. 44, no. 11, pp. 5022–5033, Jun. 2016.</br>
> https://github.com/amkozlov/sativa

vsearch
-------------------------

> "VSEARCH: a versatile open source tool for metagenomics".</br>
> Rognes et al., PeerJ, vol. 4, p. e2584, 2016.</br>
> https://github.com/torognes/vsearch

swarm
-------------------------

> "Swarm v2: highly-scalable and high-resolution amplicon clustering".</br>
> Mahé et al., PeerJ, 2015.</br>
> https://github.com/torognes/swarm

phylofactor
-------------------------

> "Phylogenetic factorization of compositional data yields lineage-level associations in microbiome datasets".</br>
> Washburne et al., PeerJ, 2017.</br>
> https://github.com/reptalex/phylofactor
